openpathsampling/netcdfplus/dictify.py

```diff
@@ -240,78 +240,71 @@ def callable_to_dict(c):
+        if root_module in ObjectJSON.allowed_imports:
+            # only store the function/class and the module
+            return {
+                '_module': c.__module__,
+                '_name': c.__name__
+            }
+
+        # if the easy way did not work, let's see if we can save it using the bytecode
+        if ObjectJSON.allow_marshal and callable(c):
+            # use marshal
+            global_vars = ObjectJSON._find_var(c, opcode.opmap['LOAD_GLOBAL'])
+            import_vars = ObjectJSON._find_var(c, opcode.opmap['IMPORT_NAME'])
+
+            builtins = dir(__builtin__)
+
+            global_vars = list(set([var for var in global_vars if var not in builtins]))
+            import_vars = list(set(import_vars))
+
+            if len(global_vars) > 0:
+                err = 'The function you try to save relies on globally set vari00ables' + \
+                      '\nand these cannot be saved since storage has no access to the' + \
+                      '\nglobal scope. This includes imports!'
+                err += '\nWe require that the following globals: ' + str(global_vars) + ' either'
+                err += '\n(1) be replaced by constants'
+                err += '\n(2) be defined inside your function,' + \
+                       '\n' + '\n'.join(map(lambda x: '    ' + x + '= ...', global_vars))
+                err += '\n(3) imports need to be "re"-imported inside your function' + \
+                       '\n' + '\n'.join(map(lambda x: '    import ' + x, global_vars))
+                err += '\n(4) be passed as an external parameter (does not work for imports!), like in '
+                err += '\n        my_cv = CV_Function("cv_name", ' + c.func_name + ', ' + \
+                       ', '.join(map(lambda x: x + '=' + x, global_vars)) + ')'
+                err += '\n    and change your function definition like this'
+                err += '\n        def ' + c.func_name + '(snapshot, ...,  ' + \
+                       ', '.join(global_vars) + '):'
+
+                raise RuntimeError('Cannot store function! Dependency on global variables. \n' + err)
+
+                # print [obj._idx for obj in global_vars if hasattr(obj, '_idx')]
+                # print [obj for obj in global_vars]
+
+            not_allowed_modules = [module for module in import_vars
+                                   if module not in ObjectJSON.allowed_imports]
+
+            if len(not_allowed_modules) > 0:
+                err = 'The function you try to save requires the following modules to ' + \
+                      '\nbe installed: ' + str(not_allowed_modules) + ' which are not marked as safe!'
+                err += '\nYou can change the list of safe modules in "CV_function._allowed_modules"'
+                err += '\nYou can also include the import startement in your function like'
+                err += '\n' + '\n'.join(['import ' + v for v in not_allowed_modules])
+
+                print err
```

Comment:
Correct. This was meant as an explanation on how to fix the issue as to be able to fix it. We can either add it to the logger or add the text to the exception. Ideas?